hela/retrieval.py

```diff
@@ -86,136 +79,83 @@ def train(self, num_trees=1000, num_jobs=5, quiet=False):
         r2scores : dict
             :math:`R^2` values for each parameter after training
         """
-        # Loading dataset
-        self.dataset = load_dataset(self.training_dataset)
-
         # Training model
-        self.model = train_model(self.dataset, num_trees, num_jobs, not quiet)
-
-        os.makedirs(self.model_path, exist_ok=True)
-        model_file = os.path.join(self.model_path, "model.pkl")
-
-        # Saving model
-        joblib.dump(self.model, model_file)
+        self.model = train_model(dataset, num_trees, num_jobs, not quiet)
 
         # saving model information...
         self.oob = self.model.rf.oob_score_
 
-        pred, r2scores = test_model(self.model, self.dataset)
+        pred, r2scores = test_model(self.model, dataset)
         self.pred = pred
         return r2scores
```

Comment:
Why don't we also return `pred`? Not saying that we must do it, but if we return r2scores, it seems natural.